dev-util/rebar/rebar-3.14.4.ebuild

```diff
@@ -0,0 +1,101 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit bash-completion-r1
+
+HEX_PACKAGES="
+bbmustache-1.10.0
+certifi-2.5.3
+cf-0.3.1
+cth_readable-1.5.0
+erlware_commons-1.4.0
+eunit_formatters-0.5.0
+getopt-1.0.1
+meck-0.8.13
+parse_trans-3.3.1
+providers-1.8.1
+relx-4.3.0
+ssl_verify_fun-1.1.6
+"
+
+S="${WORKDIR}/rebar3-${PV}"
+
+# TODO: Move to eclass.
+# @FUNCTION: hex_package_uris
+# @DESCRIPTION:
+# Generates the URIs to put in SRC_URI to help fetch dependencies from hex.pm.
+hex_package_uris() {
+	local package
+	for package in "$@"; do
+		echo "https://repo.hex.pm/tarballs/$package.tar -> $package.hex.tar"
+	done
+}
+
+# Tests currently failing.
+RESTRICT="test"
```

Comment:
Currently investigating why some of the tests are failing. An excerpt from the logs:

```
%%% rebar_dir_SUITE ==> global_cache_dir: FAILED
%%% rebar_dir_SUITE ==> 
Failure/Error: ?assertEqual([47,118,97,114,47,116,109,112,47,112,111,114,116,97,103,101,47,100,101,118,45,117,116,105,108,47,114,101,98,97,114,45,51,46,49,52,46,52,47,119,111,114,107,47,114,101,98,97,114,51,45,51,46,49,52,46,52,47,95,98,117,105,108,100,47,116,101,115,116,47,108,111,103,115,47,99,116,95,114,117,110,46,110,111,110,111,100,101,64,110,111,104,111,115,116,46,50,48,50,49,45,48,51,45,49,56,95,48,57,46,53,49,46,49,55,47,108,105,98,46,114,101,98,97,114,46,108,111,103,115,47,114,117,110,46,50,48,50,49,45,48,51,45,49,56,95,48,57,46,53,49,46,49,56,47,108,111,103,95,112,114,105,118,97,116,101,47,99,97,99,104,101], rebar_dir : global_cache_dir ( rebar_state : opts ( State ) ))
  expected: "/var/tmp/portage/dev-util/rebar-3.14.4/work/rebar3-3.14.4/_build/test/logs/ct_run.nonode@nohost.2021-03-18_09.51.17/lib.rebar.logs/run.2021-03-18_09.51.18/log_private/cache"
       got: "/var/tmp/portage/dev-util/rebar-3.14.4/temp/rebar_cache_dir"
      line: 309

%%% rebar_dir_SUITE ==> default_global_cache_dir: FAILED
%%% rebar_dir_SUITE ==> 
Failure/Error: ?assertEqual([47,118,97,114,47,116,109,112,47,112,111,114,116,97,103,101,47,100,101,118,45,117,116,105,108,47,114,101,98,97,114,45,51,46,49,52,46,52,47,104,111,109,101,100,105,114,47,46,99,97,99,104,101,47,114,101,98,97,114,51], rebar_dir : global_cache_dir ( rebar_state : opts ( State ) ))
  expected: "/var/tmp/portage/dev-util/rebar-3.14.4/homedir/.cache/rebar3"
       got: "/var/tmp/portage/dev-util/rebar-3.14.4/temp/rebar_cache_dir"
      line: 315
%%% rebar_file_utils_SUITE ==> reset_dir.reset_nonexistent_dir: FAILED
%%% rebar_file_utils_SUITE ==> 
Failure/Error: ?assert(filelib : is_dir ( TmpDir ))
  expected: false
       got: true
      line: 100
%%% rebar_file_utils_SUITE ==> reset_dir.reset_dir: FAILED
%%% rebar_file_utils_SUITE ==> 
Failure/Error: ?assert(filelib : is_file ( filename : join ( [ TmpDir , File ] ) ))
  expected: true
       got: false
      line: 121
%%% rebar_file_utils_SUITE ==> resolve_link: FAILED
%%% rebar_file_utils_SUITE ==> {{badmatch,{error,eacces}},
 [{rebar_file_utils_SUITE,resolve_link,1,
                          [{file,"/var/tmp/portage/dev-util/rebar-3.14.4/work/rebar3-3.14.4/test/rebar_file_utils_SUITE.erl"},
                           {line,193}]},
  {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1754}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1263}]},
  {test_server,run_test_case_eval,9,[{file,"test_server.erl"},{line,1195}]}]}
```

Reply:
Tests passing now, after leaving rebar_cache_dir to the default path.

```
===> Running Common Test suites...
%%% rebar_alias_SUITE: ........
%%% rebar_as_SUITE: ...........
%%% rebar_compile_SUITE: ................................................................................
%%% rebar_compile_SUITE ==> always_recompile_when_erl_compiler_options_set: SKIPPED
%%% rebar_compile_SUITE ==> {tc_user_skip,"compile:env_compiler_options/0 available"}
.....
%%% rebar_compiler_dag_SUITE: ..........
%%% rebar_compiler_epp_SUITE: ......
%%% rebar_cover_SUITE: .............
%%% rebar_ct_SUITE: ........................................................
%%% rebar_deps_SUITE: ..............................
%%% rebar_dialyzer_SUITE: ...........
%%% rebar_dir_SUITE: .......................
%%% rebar_disable_app_SUITE: .
%%% rebar_discover_SUITE: ...
%%% rebar_dist_utils_SUITE: ....
%%% rebar_edoc_SUITE: ...
%%% rebar_escriptize_SUITE: ....
%%% rebar_eunit_SUITE: .....................................
%%% rebar_file_utils_SUITE: ....................
%%% rebar_hooks_SUITE: ...........
%%% rebar_install_deps_SUITE: .......................................
%%% rebar_lock_SUITE: ....
%%% rebar_namespace_SUITE: ...............
%%% rebar_new_SUITE: ......
%%% rebar_opts_parser_SUITE: ..
%%% rebar_paths_SUITE: ....
%%% rebar_pkg_SUITE: ...........
%%% rebar_pkg_alias_SUITE: ....
%%% rebar_pkg_repos_SUITE: .............
%%% rebar_plugins_SUITE: ..........
%%% rebar_profiles_SUITE: .........................
%%% rebar_release_SUITE: ............
%%% rebar_resource_SUITE: ...
%%% rebar_src_dirs_SUITE: ............
%%% rebar_templater_SUITE: ..
%%% rebar_unlock_SUITE: ...
%%% rebar_upgrade_SUITE: .............................................................
%%% rebar_uri_SUITE: ..
%%% rebar_utils_SUITE: ............................
%%% rebar_xref_SUITE: ....
Skipped 1 (1, 0) tests. Passed 596 tests.
>>> Completed testing dev-util/rebar-3.14.4
```